src/lib/utils/tests/test_validation.py

```
@@ -0,0 +1,268 @@
+            self.assertEqual(result, os.path.abspath(tmpdir))
+
+    def test_nonexistent_path_raises(self):
+        with self.assertRaises(osmo_errors.OSMOUserError):
+            validation.valid_path('/nonexistent/path/xyz/123')
+
+
+class TestValidFilePath(unittest.TestCase):
+    """ Tests for valid_file_path. """
+
+    def test_nonexistent_path_returned_as_is(self):
+        path = '/tmp/this-file-should-not-exist-xyz123'
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/this-file-should-not-exist-xyz123"

(S108)